src/mono/mono/mini/mini-amd64.c

```diff
@@ -3882,8 +3882,16 @@ mono_arch_lowering_pass (MonoCompile *cfg, MonoBasicBlock *bb)
 		case OP_XEQUAL: {
 			int temp_reg1 = mono_alloc_ireg (cfg);
 			int temp_reg2 = mono_alloc_ireg (cfg);
+			int opcode= -1;
+			switch (ins->inst_c1)
+			{
+			case MONO_TYPE_R4: opcode = OP_COMPPS; break;
+			case MONO_TYPE_R8: opcode = OP_COMPPD; break;
+			default: opcode = OP_PCMPEQD;
+			}
+			ins->inst_c0 = 0;
```

Comment:
Is this needed?

Reply:
I think that yes because the `ins->inst_c0` is used to specify to compare for equality as here https://github.com/dotnet/runtime/blob/b07a2f2c5a72561d644202bf897a3b06cbce3080/src/mono/mono/mini/mini-amd64.c#L6793-L6796

Reply:
Will the `ins` be passed down to the compare op code?

Reply:
You are right, it should probably by `temp->inst_c0 = 0` after this line https://github.com/dotnet/runtime/blob/6d8e652cbd69aa94e4c4509d80db1b4b68539bc2/src/mono/mono/mini/mini-amd64.c#L3894.

Reply:
The default value for `temp->inst_c0` is 0. There is no need to set it to zero explicitly.